rollout/rollout.go

```diff
@@ -122,14 +123,28 @@ func (r *Runner) Execute(rolloutConfig *Config) (err error) {
 	}
 
 	for _, group := range res.Rollout.Spec.Groups {
+		var wg sync.WaitGroup
+		errs := make(chan error, len(group.Steps))
 		for _, step := range group.Steps {
-			object, found := res.Objects[step.Object]
-			if !found {
-				return fmt.Errorf("Could not find object named %q", step.Object)
-			}
+			wg.Add(1)
 
-			err := r.runStep(step, object)
-			if err != nil {
+			go func(step *types.Step) {
+				defer wg.Done()
+
+				object, found := res.Objects[step.Object]
+				if !found {
+					errs <- fmt.Errorf("could not find object named %q", step.Object)
+					return
+				}
+				if err := r.runStep(step, object); err != nil {
+					errs <- err
+					return
+				}
+			}(step)
+
+			wg.Wait()
```

Comment:
Maybe I'm reading this incorrectly but shouldn't this wait have to be one upper scope? (outer for)

Reply:
Wow. So right. Oops.